e2e/testcases/helm_sync_test.go

```diff
@@ -380,3 +409,75 @@ func helmChartVersion(chartVersion string) nomostest.Sha1Func {
 		return chartVersion, nil
 	}
 }
+
+// pushes a new helm chart for use during an e2e test
+func pushHelmChart(nt *nomostest.NT, t *testing.T, helmchart, version string) string {
+	t.Helper()
+
+	nt.T.Log("Push helm chart to the artifact registry")
+	tmpDir := t.TempDir()
+
+	chartName := fmt.Sprintf("%s-%s-%s-%s", helmchart, nomostesting.GCPProjectIDFromEnv, nomostesting.GCPClusterFromEnv, generateRandomString())
+	if len(chartName) > 50 {
+		// the chartName + releaseName is used as the metadata.name of resources in the coredns helm chart, so we must trim this down
+		// to keep it under the k8s length limit
+		chartName = chartName[len(chartName)-50:]
+		chartName = strings.Trim(chartName, "-")
+	}
+
+	paths, err := filepath.Glob(fmt.Sprintf("../testdata/helm-charts/%s/*", helmchart))
```

Comment:
This fits with the existing layout of statically provisioned helm charts, but I'd like to see this follow a similar pattern as the [Git abstraction](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/gitproviders/repository.go#L561).

i.e. the test adds files to the local repo/workspace and then pushes the artifact to the remote sync source (git server in the case of git, helm registry in the case of helm).

I'd be okay with going this route for now and building on this in the future, but I'd like to at least set up these interfaces/abstractions so that it will be easy to make that change

Reply:
Yeah that makes sense. I added a new abstraction layer under `nomostest` for this, please feel free to leave feedback on it